ma/maInput.h

```diff
@@ -131,7 +124,256 @@ class Input
     const char* debugFolder;
 };
 
+
+/** \brief User configuration for a MeshAdapt run */
+class Input
+{
+  public:
+    virtual ~Input();
+    Input(Input* in) {this->ops = in->ops;}
+
+/** \brief generate an Input based on an anisotropic function.
+ \param s if non-zero, use that to transfer all fields. otherwise,
+          transfer any associated fields with default algorithms
+ \param logInterpolation if true uses logarithmic interpolation */
+    Input(
+      Mesh* m,
+      AnisotropicFunction* f,
+      SolutionTransfer* s=0,
+      bool logInterpolation=true);
+/** \brief generate an Input based on an isotropic function
+ \param s if non-zero, use that to transfer all fields. otherwise,
+          transfer any associated fields with default algorithms */
+    Input(
+      Mesh* m,
+      IsotropicFunction* f,
+      SolutionTransfer* s=0);
+/** \brief generate an Input based on anisotropic fields
+ \param sizes a vector field of desired element sizes along the
+              axes of the anisotropic frame
+ \param frames a matrix field containing anisotropic frames
+               for each vertex
+ \param s if non-zero, use that to transfer all fields. otherwise,
+          transfer any associated fields with default algorithms
+ \param logInterpolation if true uses logarithmic interpolation */
+    Input(
+      Mesh* m,
+      apf::Field* sizes,
+      apf::Field* frames,
+      SolutionTransfer* s=0,
+      bool logInterpolation=true);
+/** \brief generate an Input based on an isotropic field
+ \param size a scalar field of desired element size
+ \param s if non-zero, use that to transfer all fields. otherwise,
+          transfer any associated fields with default algorithms */
+    Input(
+      Mesh* m,
+      apf::Field* size,
+      SolutionTransfer* s=0);
+
+/** \brief generate a uniform refinement Input */
+    Input(Mesh* m, int n=1, SolutionTransfer* s=0);
+/** \brief generate a no-op Input */
+    Input(Mesh* m, SizeField* f=0, SolutionTransfer* s=0);
+
+    // setters and getters
+/** \brief number of refine/coarsen iterations to run (default 3)
+    \details this value will be set to the minimum required iterations
+    inside the call to ma::configure in cases where there is a size information.
+    Users can override this by setting in->maximumIterations after the
+    call to ma::configure and before the call to ma::adapt routine.*/
+    int maximumIterations();
```

Comment:
all of these getter functions should be const (and noexcept if we use c++11) qualified since they don't modify the class.
i.e. `int maximumIterations() const noexcept`